controlplane/kubeadm/internal/controllers/remediation.go

```diff
@@ -67,6 +71,11 @@ func (r *KubeadmControlPlaneReconciler) reconcileUnhealthyMachines(ctx context.C
 		return ctrl.Result{}, kerrors.NewAggregate(errList)
 	}
 
+	// Returns if another remediation is in progress but the new machine is not yet created.
+	if _, ok := controlPlane.KCP.Annotations[controlplanev1.RemediationInProgressAnnotation]; ok {
```

Comment:
Just wanted to note that after a successful remediation there is no guarantee that the controller will see that the annotation is gone because the cache could be stale (hit this directly when testing locally)